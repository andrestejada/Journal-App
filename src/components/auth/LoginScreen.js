import React from 'react';
import {useDispatch, useSelector} from 'react-redux'
import {Link} from 'react-router-dom';
import { loginWithGoogle, starLoginEmailAndPassword } from '../../actions/auth';
import { setError } from '../../actions/ui';
import { useForm } from '../../hooks/UseForm';



const LoginScreen = () => {
    
    const [values, handleInputChange,]=useForm({
        email:'luis@correo.com',
        password:'luchito'
    });

    const { email,password}=values;
    const dispatch = useDispatch(); 
    const {msgError , loading} = useSelector(state => state.ui);

    //submit del login
    const handleSubmit=(e)=>{
        e.preventDefault();
        //validar
        if( email.trim()===''|| password.trim()==='' ){
            dispatch( setError('Ambos campos son obligatorios') );
            return; 
        }
        ///paso la validacion 
        dispatch( starLoginEmailAndPassword(email,password) );
    }

    const handleLoginClick =()=>{
        dispatch( loginWithGoogle() )
    }
    return (
        <>
            <h3 className='auth__title' >Login</h3>
            {
            msgError &&
            <div className='auth__alert-error' >{msgError}</div>
            }
            <form
               onSubmit={handleSubmit}
               className="animate__animated animate__fadeIn animate__faster" 
            >
                <input
                    type='text'
                    name='email'
                    placeholder='correo@correo.com'
                    className='auth__input'
                    value={email}
                    onChange={handleInputChange}
                />
                <input
                    type='password'
                    name='password'
                    placeholder='******'
                    className='auth__input'
                    value={password}
                    onChange={handleInputChange}
                />
                <button
                    type='submit'
                    className='btn btn-primary btn-block'
                    disabled={ loading }
                >Ingresar</button>
                <hr/>
                <div className='auth__social-networks' >
                    <p>Login con Redes Sociales</p>
                    <div 
                        className="google-btn"
                        onClick={handleLoginClick}
                    >
                        <div className="google-icon-wrapper">
                            <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
                        </div>
                        <p className="btn-text">
                            <b>Sign in with google</b>
                        </p>
                    </div>
                </div>
                <Link
                    to='/auth/register'
                    className='link'
                >Crear Nueva Cuenta</Link>
            </form>
        </>
    )
}

export default LoginScreen