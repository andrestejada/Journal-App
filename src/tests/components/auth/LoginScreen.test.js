import React from 'react';
import LoginScreen from '../../../components/auth/LoginScreen'
import {MemoryRouter} from 'react-router-dom'
import {mount } from 'enzyme'
import {Provider} from 'react-redux'
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import '@testing-library/jest-dom'




const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState={
    auth:{},
    ui:{
        loading:false,
        msgError:null
    }
}

let store = mockStore(initState)

const wrapper = mount( 
    <Provider store={store} >
        <MemoryRouter>
            <LoginScreen/> 
        </MemoryRouter>
    </Provider>
)
 

describe('Pruebas del component <LoginScreen/>', () => {
    beforeEach(()=>{
        store = mockStore(initState)
    });
    test('se debe de renderizar correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();
    });
});