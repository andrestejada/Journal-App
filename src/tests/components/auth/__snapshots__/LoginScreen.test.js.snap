// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del component <LoginScreen/> se debe de renderizar correctamente 1`] = `ReactWrapper {}`;
